src/main/java/net/mcreator/plugin/events/ui/TabEvent.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2022, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.plugin.events.ui;
+
+import net.mcreator.plugin.MCREvent;
+import net.mcreator.ui.MCreatorTabs;
+
+/**
+ * <p>Events for different actions of a {@link MCreatorTabs.Tab} component.
+ * All events are triggered BEFORE the action is executed.</p>
+ */
+public class TabEvent extends MCREvent {
+
+	private final MCreatorTabs.Tab tab;
+
+	public TabEvent(MCreatorTabs.Tab tab) {
+		this.tab = tab;
+	}
+
+	public MCreatorTabs.Tab getTab() {
+		return tab;
+	}
+
+	/**
+	 * <p>Triggered when a {@link MCreatorTabs.Tab} is added to the UI.</p>
+	 */
+	public static class AddTabEvent extends TabEvent {

```

Comment:
This can be add only, or better, added.

Right now it reads:

`TabEvent.AddTabEvent`

but it would be better as

`TabEvent.Added`

or maybe

`TabEvent.TabAdded`

Reply:
Similar with others in this class